packages/eslint-plugin/src/stylex-margin-padding-shorthand.js

```diff
@@ -72,143 +80,161 @@ const shorthandAliases = {
+      noLegacyFormats:
+        'Use {{correctProperty}} instead of legacy formats like {{incorrectProperty}} to adhere to logical property naming.',
     },
-    create(context: Rule.RuleContext): { ... } {
-      function validateObject(obj: ObjectExpression) {
-        for (const prop of obj.properties) {
-          if (prop.type === 'SpreadElement') {
-            continue;
-          }
-          if (prop.value.type === 'ObjectExpression') {
-            validateObject(prop.value);
-          } else {
-            validateProperty(prop);
-          }
-        }
-      }
-  
-      function validateProperty(property: Property) {
-        if (property.computed) {
-          // can't resolve computed keys
-          return;
-        }
-  
-        let key;
-        if (property.key.type === 'Identifier') {
-          key = property.key.name;
-        } else if (property.key.type === 'Literal') {
-          key = property.key.value;
-        }
-  
-        if (typeof key === 'string' && key in legacyNameMapping) {
-          context.report({
-            node: property,
-            message: 'Use {{correctProperty}} instead of legacy formats like {{incorrectProperty}} to adhere to logical property naming.',
-            data: {
-              correctProperty: legacyNameMapping[key],
-              incorrectProperty: key,
-            },
-            fix: fixer => {
-              // $FlowFixMe - We've already checked that key is a string and in legacyNameMapping
-              return fixer.replaceText(property.key, legacyNameMapping[key]);
-            },
-          });
+    fixable: 'code',
+  },
+  create(context: Rule.RuleContext): { ... } {
+    function validateObject(obj: ObjectExpression) {
+      for (const prop of obj.properties) {
+        if (prop.type === 'SpreadElement') {
+          continue;
         }
-  
-        if (typeof key !== 'string' || !shorthandAliases.hasOwnProperty(key) || property.value.value === null) {
-          return;
+        if (prop.value.type === 'ObjectExpression') {
+          validateObject(prop.value);
+        } else {
+          validateProperty(prop);
         }
+      }
+    }
+
+    function validateProperty(property: Property) {
+      if (property.computed) {
+        // can't resolve computed keys
+        return;
+      }
 
-        const newValues = shorthandAliases[key](property.value.value);
-  
+      let key;
+      if (property.key.type === 'Identifier') {
+        key = property.key.name;
+      } else if (property.key.type === 'Literal') {
+        key = property.key.value;
+      }
+
+      if (typeof key === 'string' && key in legacyNameMapping) {
         context.report({
           node: property,
-          message: `Property shorthands like ${key}: 0 0 0 0 are not supported in styleX. Separate into ${key}Start, ${key}End [...].`,
+          message:
+            'Use {{correctProperty}} instead of legacy formats like {{incorrectProperty}} to adhere to logical property naming.',
```

Comment:
```suggestion
            `Use "${correctProperty}" instead of legacy formats like "${incorrectProperty}" to adhere to logical property naming.',
```